cmd/templ/generatecmd/main.go

```diff
@@ -129,15 +147,36 @@ func runCmd(ctx context.Context, w io.Writer, args Arguments) (err error) {
 		p = proxy.New(args.ProxyPort, target)
 	}
 	fmt.Fprintln(w, "Processing path:", args.Path)
+
+	if args.Watch {
+		err = generateWatched(watchCtx, w, args, opts, p)
+		if err != nil && !errors.Is(err, context.Canceled) {
+			return err
+		}
+	}
+
+	return generateProduction(ctx, w, args, opts, p)
+}
+
+func generateWatched(ctx context.Context, w io.Writer, args Arguments, opts []generator.GenerateOpt, p *proxy.Handler) error {
+	fmt.Fprintln(w, "Generating dev code:", args.Path)
+	start := time.Now()
+
 	bo := backoff.NewExponentialBackOff()
 	bo.InitialInterval = time.Millisecond * 500
 	bo.MaxInterval = time.Second * 3
 	bo.MaxElapsedTime = 0
 
 	var firstRunComplete bool
 	fileNameToLastModTime := make(map[string]time.Time)
+	fileNameToHash := make(map[string][md5.Size]byte)
```

Comment:
We're not using MD5 for security here, but use of it will flag as a deprecated / insecure hashing algorithm in security reviews, so I'd prefer to use SHA256.

I doubt there's much difference in the performance of MD5 vs SHA256, given that Go uses specific processor instructions for architectures that support it.

https://cs.opensource.google/go/go/+/master:src/crypto/sha256/sha256block_amd64.go

https://cs.opensource.google/go/go/+/master:src/crypto/sha256/sha256block_arm64.go